src/openai/_models.py

```diff
@@ -363,7 +363,12 @@ class GenericModel(BaseGenericModel, BaseModel):
 
 
 if PYDANTIC_V2:
-    from pydantic import TypeAdapter
+    from functools import lru_cache
+    from pydantic import TypeAdapter as PyTypeAdapter
+
+    @lru_cache
```

Comment:
The default `maxsize` is `128`, if we're going to do this, would that cause any issues? should it be higher?

Reply:
I don't think it would be a problem, here it's just caching each unnecessary instantiation (as mentioned in the docs), thus improving performance. Excessively large values also take up more memory.